lib/reevoo-ping.js

```diff
@@ -12,13 +12,20 @@ import Badge                  from './events/badge';
 import Conversion             from './events/conversion';
 
 class Client {
-  constructor(appId, opts = {}, trackerConfig = {}) {
+  constructor(appId, opts = {}, trackerConfig = {}, win = window) {
     if (!appId) throw new Error('appId argument is required');
 
     // Initialise the tracker
     const snowplowTrackerConfig = { ...defaultSnowplowConfig, ...trackerConfig, appId };
     snowplow('newTracker', appId, COLLECTOR_URI, snowplowTrackerConfig);
 
+    // Snowplow collector cannot handle URLs with characters violating RFC 2396, if customer URL contains some of
+    // them we set our custom encoded version to be tracked.
+    const encodedURI = encodeURI(win.location.href).replace(/!/g, '%21');
+    if (encodedURI !== win.location.href) {
+      snowplow('setCustomUrl', encodedURI);
```

Comment:
Yes, it could be set for all but I want to be super careful as it's really difficult to reprocess those millions of events if something goes wrong.